File: SW Project/Customised online judger/problem_set/2017/matrix.java
// Arup Guha
// 7/31/2013
// Solution to 2011 UCF Locals Problem: Matrix
import java.util.*;

public class matrix {

	public static void main(String[] args) {

		Scanner stdin = new Scanner(System.in);
		int numCases = stdin.nextInt();

		// Go through each case.
		for (int loop=1; loop<=numCases; loop++) {

			// Store sums of "even" and "odd" indexes.
			int r = stdin.nextInt();
			int c = stdin.nextInt();
			int evenSum = 0, oddSum = 0;
			for (int i=0; i<r; i++) {
				for (int j=0; j<c; j++) {
					int val = stdin.nextInt();
					if ( (i+j)%2 == 0)
						evenSum += val;
					else
						oddSum += val;
				}
			}

			// Output result - this is the invariant in the given
			// transformation...
			if (evenSum == oddSum)
				System.out.println("0");
			else
				System.out.println("1");
		}
	}
}

File: SW Project/Customised online judger/problem_set/2017/matrix.py
# Arup Guha
# 8/8/2017
# Solution to 2011 UCF Locals Problem Matrix - ported from Java.

numCases = int(input(""))

# Process each case.
for loop in range(1,numCases+1):

    # Read in dimensions.
    toks = input("").split()
    r = int(toks[0])
    c = int(toks[1])
    evenSum = 0
    oddSum = 0

    # Read in row by row.
    for i in range(r):
        toks = input("").split()

        # Add each item to the appropriate sum.
        for j in range(c):
            item = int(toks[j])
            if (i+j)%2 == 0:
                evenSum += item
            else:
                oddSum += item
                
    # Ta da!
    if evenSum == oddSum:
        print("0")
    else:
        print("1")
